python/custom_config_reader.py

```diff
@@ -0,0 +1,48 @@
+import attr
+from looker_sdk import (
+    api_settings,
+    auth_session,
+    methods,
+    requests_transport,
+    serialize,
+)
+from looker_sdk.rtl import transport as tp
+
+
+@attr.s(auto_attribs=True, kw_only=True)
+class CustomConfigReader(tp.TransportSettings):
+    """
+    A custom configuration reader that implements the PApiSettings protocol and allows
+    passing of config details as parameters. The get_client_id and get_cient_secret
+    methods can be modified to fetch credentials from anywhere.
+    """
+
+    @classmethod
+    def configure(cls, base_url: str, api_version: str) -> api_settings.PApiSettings:
+        settings = cls(base_url=base_url, api_version=api_version)
+        return settings
+
+    def get_client_id(self) -> str:
+        return "client_id"
+
+    def get_client_secret(self) -> str:
+        return "client_id"
```

Comment:
ug, so this was a discrepancy that needs to be fixed. `typescript/looker/rtl/apiSettings.ts` exposes `readConfig` as opposed to `get_client_id` and `get_client_secret`. That's because we're going to have other means (like the new OAuth stuff). We should implement extracting the client id/secret inside the auth session login.

Let's hold off on merging this till I fix that up. Then this example will implement `read_config()` and put `"client_id"` and `"client_secret"` keys into the "bag"